packages/core/src/state-management/state-manager.ts

```diff
@@ -442,17 +454,28 @@ export class StateManager {
             }
             case AnchorRequestStatusName.COMPLETED: {
               await this._handleAnchorCommit(state$, asr.cid, asr.anchorCommit.cid, asr.witnessCar)
-              await this.anchorRequestStore.remove(state$.id)
+
+              if (!this._tipHasProcessingAnchorRequest(state$)) {
+                await this.anchorRequestStore.remove(state$.id)
+              }
               stopSignal.next()
               return
             }
             case AnchorRequestStatusName.FAILED: {
-              if (!asr.cid.equals(state$.tip)) return
               this.logger.warn(
                 `Anchor failed for commit ${asr.cid} of stream ${asr.streamId}: ${asr.message}`
               )
-              state$.next({ ...state$.value, anchorStatus: AnchorStatus.FAILED })
-              await this.anchorRequestStore.remove(state$.id)
+
+              // if this is the anchor response for the tip update the state
+              if (asr.cid.equals(state$.tip)) {
+                state$.next({ ...state$.value, anchorStatus: AnchorStatus.FAILED })
+              }
+
+              if (!this._tipHasProcessingAnchorRequest(state$)) {
```

Comment:
so is this just an extra safety check just in case?  Is there ever a scenario we would expect this to actually run?

Reply:
The scenario could be:
We have CommitA requested. CommitA is PROCESSING. Before waiting for CommitA to complete or fail we request CommitB. If CommitA fails we don't want to remove the stream from the anchorRequestStore because CommitB might pass. 


If CommitA passes we would be fine removing the stream from the anchorRequestStore because the anchor for CommitB will be invalid anyway. But that will make the logic very confusing.

Reply:
I guess I meant but where `asr.cid.equals(state$.tip)` is false.  I'm wondering why we need any changes to the `FAILED` case at all.  It seems like the case where `this._tipHasProcessingAnchorRequest` is true would always require that `asr.cid.equals(state$.tip)` be false, in which case with the original logic we would have done nothing here anyway

Reply:
> It seems like the case where this._tipHasProcessingAnchorRequest is true would always require that asr.cid.equals(state$.tip) be false,

Can state$ have been updated in between `asr.cid.equals(state$.tip)` and `this._tipHasProcessingAnchorRequest`.... Maybe but then this doesn't fix anything anyway. The more that i think about this the more I think any approach other than DB transaction won't work. 

I'mg oing to reduce the scope of this PR to things that can actually be fixed and file the ticket for the rest